services/provider/server/server.go

```diff
@@ -596,5 +526,119 @@ func (s *OCSProviderServer) RevokeStorageClassClaim(ctx context.Context, req *pb
 
 // GetStorageClassClaim RPC call to get the ceph resources for the StorageclassClaim.
 func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req *pb.StorageClassClaimConfigRequest) (*pb.StorageClassClaimConfigResponse, error) {
-	return &pb.StorageClassClaimConfigResponse{}, nil
+	consumerClaimObj, err := s.storageClassClaimManager.Get(ctx, req.StorageConsumerUUID, req.StorageClassClaimName)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "failed to get storage class claim config %q for %q. %v", req.StorageClassClaimName, req.StorageConsumerUUID, err)
+		}
+		return nil, status.Errorf(codes.Internal, "failed to get storage class claim config %q for %q. %v", req.StorageClassClaimName, req.StorageConsumerUUID, err)
+	}
+
+	// TODO: add status.Phase validation here to check claim is in expected
+	// state or not before extracting the ceph resources from the StorageClassClaim CR status
+	conString, err := s.getExternalResourcesFromClaim(ctx, consumerClaimObj)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get external resources. %v", err)
+	}
+
+	return &pb.StorageClassClaimConfigResponse{ExternalResource: conString}, nil
+}
+
+func (s *OCSProviderServer) getExternalResourcesFromClaim(ctx context.Context, storageClassClaim *ocsv1alpha1.StorageClassClaim) ([]*pb.ExternalResource, error) {
+	var extR []*pb.ExternalResource
+	for _, cephRes := range storageClassClaim.Status.CephResources {
+		switch cephRes.Kind {
+		case "CephClient":
+			clientSecretName, _, _, err := s.getCephClientInformation(ctx, cephRes.Name)
```

Comment:
After these changes, the second return value is always `_` in all call sites. 
If we don't need it, let's not return it in the first place.

In addition, the third return value is also meaningless, in this function it is not used and in the original function, it will always be the same value as we are always dealing with the health checker client.
So let's remove it as well and rename the util to `getCephClientSecretName`

Reply:
removed yet, I didn't optimize anything from previously written functions.

Reply:
Ok